astropy/units/physical.py

```diff
@@ -1,133 +1,425 @@
+    °Ra. However, each of these temperatures each represents the same
+    physical type.  Replacing the different temperature units with
+    kelvin allows the physical type to be treated consistently.
+    """
+    physical_type_id = unit._get_physical_type_id()
+
+    physical_type_id_components = []
+    substitution_was_made = False
+
+    for base, power in physical_type_id:
+        if base in ["deg_F", "deg_C", "deg_R"]:
+            base = "K"
+            substitution_was_made = True
+        physical_type_id_components.append((base, power))
+
+    if substitution_was_made:
+        return core.Unit._from_physical_type_id(tuple(physical_type_id_components))
+    else:
+        return unit
+
+
+class PhysicalType:
     """
-    Adds a new physical unit mapping.
+    Represents and provides information on the physical type(s) that are
+    associated with a set of units.
+
+    .. warning::
+        This class is not intended to be instantiated by the user.
 
     Parameters
     ----------
-    unit : `~astropy.units.UnitBase` instance
-        The unit to map from.
+    unit : u.Unit
+        The unit to be represented by the physical type.
+
+    physical_types : `str` or `set` of `str`
+        A `str` representing the name of the physical type of the unit,
+        or a `set` containing strings that represent one or more names
+        of physical types.
+
+    Examples
+    --------
+    The preferred method to access a physical type is by accessing the
+    ``physical_type`` attribute of a unit.
+
+    >>> import astropy.units as u
+    >>> u.meter.physical_type
+    'length'
+
+    Occasionally, a set of units will correspond to more than one
+    physical type.
+
+    >>> u.Pa.physical_type
+    {'energy density', 'pressure'}
+    >>> u.Pa.physical_type == 'pressure'
+    True
+    >>> 'energy density' in u.Pa.physical_type
+    True
+
+    Physical types may be used for dimensional analysis.
+
+    >>> length = u.pc.physical_type
+    >>> area = (u.cm ** 2).physical_type
+    >>> length * area
+    'volume'
 
-    name : str
-        The physical name of the unit.
+    Multiplcation, division, and exponentiation work with physical types.
+
+    >>> length * area
+    'volume'
+    >>> area / length
+    'length'
+    >>> length ** 3
+    'volume'
+
+    Unknown physical types will be labelled as ``"unknown"``.
+
+    >>> (u.s ** 13).physical_type
+    'unknown'
+
+    Dimensional analysis may be performed for unknown physical types too.
+
+    >>> length_to_19th_power = (u.m ** 19).physical_type
+    >>> length_to_20th_power = (u.m ** 20).physical_type
+
+    >>> length_to_19th_power, length_to_20th_power
+    ('unknown', 'unknown')
+
+    >>> length_to_20th_power / length_to_19th_power
+    'length'
+
+    A string representation of a physical type can include underscores
+    instead of spaces, and multiple physical types in a single string
+    can be separated by a comma or backslash.
```

Comment:
> It is a regular slash.

I've been getting this wrong probably my whole life!

> But I wonder if it isn't better to just given one way only to indicate multiple possibilities (if so, I'd go with comma, though I could live with `|` - which I see you allow too).

The regular slash was there for backwards compatibility, since some physical types were defined like `"momentum/impulse"`.  I'm wondering if it'd be best to leave that in there for backwards compatibility, but not to describe it as part of the API so it's okay to change later.  In the interest of there being one and only one way to obviously do it, perhaps it'd be best to say that it has to be put in as a `set`.  I'll keep thinking...

> Similarly, I'm not sure about allowing underscores, but even if you do, I'd not mention it.

The underscores are something that could make my life a little bit easier later on.  I'm in the process of developing classes to represent normalizations for different sets of equations that describe plasmas.  The attributes of these classes will have the same names as some of the physical types, except with underscores instead of spaces, so allowing underscores would allow having a variable name represent a physical type.  I'd still like to mention it since there's a use case, though I can make the mention brief.

Reply:
Definitely not a big deal! If you have a possible use for it, go for it. For the slash, I think backwards compatibility is definitely important. We could simply state that `/` is supported for backwards compatibility.